internal/dependency-track/setup.sh

```diff
@@ -0,0 +1,157 @@
+#!/bin/bash
+
+# Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+######################################################################
+##   This script will set up the environment for Dependency Track   ##
+######################################################################
+
+# ---------------
+# Update Packages
+# ---------------
+
+sudo apt update -y
+sudo apt upgrade -y
+
+# -------------
+# Install MySQL
+# -------------
+
+sudo apt install -y mysql
+
+read -p "Enter Password for MySQL user 'dependency-track': "  dependency_track_password
+sudo mysql <<END
+  CREATE DATABASE dependency_track;
+  USE dependency_track;
+  CREATE USER 'dependency-track'@'localhost' IDENTIFIED BY '$dependency_track_password';
+  GRANT GRANT ALL PRIVILEGES ON dependency_track.* TO 'dependency-track'@'localhost';
+  FLUSH PRIVILEGES;
+END
+
+# -------------
+# Install Nginx
+# -------------
+
+sudo apt install -y nginx
+
+# ---------------------
+# Add Application Users
+# ---------------------	
+
+sudo useradd dependency-track
+
+# --------------
+# Setup Firewall
+# --------------
+
+sudo ufw allow 'Nginx HTTP'
+sudo ufw allow 'Nginx HTTPS'
+sudo ufw allow 'Nginx Full'
+sudo ufw allow 'OpenSSH'
+# To be removed after Nginx setup is complete
+sudo ufw allow 8080/tcp
+sudo ufw status verbose
+sudo ufw enable
+
+# -----------------------------
+# Install SDKMan, JDK and Maven
+# -----------------------------
+
+sudo su -
+
+export SDKMAN_DIR="/usr/local/sdkman" && curl -s "https://get.sdkman.io" | bash - 
+
+cat <<END >> /home/dependency-track/.bashrc
+export SDKMAN_DIR="/usr/local/sdkman"
+[[ -s "/usr/local/sdkman/bin/sdkman-init.sh" ]] && source "/usr/local/sdkman/bin/sdkman-init.sh"
+END
+
+sdk install java 8.0.212-amzn
+sdk install maven 3.6.1
+
+# -----------------------------
+# Change SQL Mode
+# -----------------------------
+
+sudo cat <<END >> /home/dependency-track/etc/mysql/my.cnf
+[mysqld]
+sql_mode="ANSI_QUOTES,STRICT_TRANS_TABLES,ONLY_FULL_GROUP_BY,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION"
+END
+
+sudo service mysql restart
+
+# -----------------------------
+# Get MySQL connector
+# -----------------------------
+
+mkdir libs
```

Comment:
IMU, we are running all these commands from root user. Hence, the `lib`, etc will get created in root's home. 

Should we `su dependency-track` and cd into `/home/dependency-track` before doing this, so that application artifacts will be created in the correct user folder?

Reply:
Done